public/validate-event-form.js

```diff
@@ -0,0 +1,121 @@
+var AddEventForm = document.getElementById('AddEventForm');
+var EditEventForm = document.getElementById('EditEventForm');
+var errorBox = document.getElementById('EventErrorBox');
+
+var englishChars = /[a-zA-Z0-9]/;
+var arabicChars = /[ا-ي0-9]/;
+
+function NotValid(message) {
+  errorBox.innerText = message;
+  window.scrollTo(0, 0);
+}
+
+function Validate(event) {
+  event.preventDefault();
+  var elements = event.target.elements;
+
+  var data = {
+    name_en: elements.name_en.value,
+    name_ar: elements.name_ar.value,
+    description_en: elements.description_en.value,
+    description_ar: elements.description_ar.value,
+    eventPlace: elements.eventPlace.value,
+    categories: elements.categories,
+    startTime: elements.startTime.value,
+    endTime: elements.endTime.value,
+    accessibility: elements.accessibility,
+    imageUrl: elements.imageUrl.value,
+    cost: elements.cost.value,
+  };
+
+  // Check if event name input were filled
+  if (!data.name_ar && !data.name_en) {
+    return NotValid('*Please input a name');
+  } else {
+    errorBox.innetText = '';
+  }
+
+  // Validate what language was used for each name input
+
+  if (data.name_en && !englishChars.test(data.name_en)) {
+    return NotValid('*Only use english characters for english event name');
+  } else {
+    errorBox.innerText = '';
+  }
+
+  if (data.name_ar && !arabicChars.test(data.name_ar)) {
+    return NotValid('*Only use arabic characters for arabic event name');
+  } else {
+    errorBox.innerText = '';
+  }
+
+  // Check if event description input were filled
+  if (!data.description_en && !data.description_ar) {
+    return NotValid('*Please write a description about the event');
+  } else {
+    errorBox.innetText = '';
+  }
+
+  // Validate what language was used for each description input
+  if (data.description_en && !englishChars.test(data.description_en)) {
+    return NotValid(
+      '*Only use english characters for english event description',
+    );
+  } else {
+    errorBox.innerText = '';
+  }
+
+  if (data.description_ar && !arabicChars.test(data.description_ar)) {
+    return NotValid('*Only use arabic characters for arabic event description');
+  }
+
+  // Check if the user has selected a place
+  if (!data.place) {
+    NotValid('*Please select a location/place');
+  } else {
+    errorBox.innerText = '';
+  }
+
+  // Check if a category was selected
+  var categorySelectedCheck = false;
+  var checkedCategories = [];
+  data.categories.forEach(function(categoryElement) {
```

Comment:
see comments below on using a filter. Should have a `filter` then a `map` here

Reply:
then `categorySelectedCheck` can be found with a reduce

Reply:
so took me a while to figure out why my code wasnt working with reduce or filter or map.
and it's because `data.categories` is a nodeList and i am only allowed do a forEach